### PR TITLE
[RATOM-84] record nonrecord

### PR DESCRIPTION
**Pull request by michael-caktus**
1. Updates MessageAudit with needs_redaction and is_restricted booleans.
2. Adds MessageAudit serializer with custom `update` method 
3. Adds audit to message serializer
4. Adds PUT method to `message_detail` endpoint. This handles MessageAudit updates.
5. Adds post_save signal to catch MessageAudit saves and update the messages index accordingly
6. Tests signals and view/serializer behavior

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+def mock_es_registry():
+    """Fixture to mock ES registry and use it automatically in every test."""
+    with mock.patch("django_elasticsearch_dsl.signals.registry") as mock_registry:
+        yield mock_registry
+
+
+@pytest.fixture(scope="session", autouse=True)
+def mock_core_registry():
+    """Fixture to mock ES registry from core.signals and use it automatically in every test."""
+    with mock.patch("core.signals.registry") as mock_core_registry:
```
Comment (michael-caktus) [inline conftest.py:17]: Nice!

Comment (JeremyGibson) [reply conftest.py:17]: Schweet.

File: api/tests/test_signals.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import pytest
+
+
+@pytest.mark.django_db
+def test_message_audit_update_triggers_message_reindex(
+    ratom_message_audit, mock_core_registry
```
Comment (JeremyGibson) [inline api/tests/test_signals.py:7]: This can be removed now, right?

Comment (JeremyGibson) [reply api/tests/test_signals.py:7]: the `mock_core_registry`

File: api/views/message.py [hunk 3]
```diff
@@ -35,6 +35,18 @@ def message_detail(request, pk):
...
+    if request.method == "PUT":
+        """
+        We don't really edit messages-- this endpoint updates an associated MessageAudit
+        """
+        serialized_audit = MessageAuditSerializer(
+            message.audit, data=request.data, partial=True
+        )
+        if serialized_audit.is_valid():
+            serialized_audit.save(updated_by=request.user)
+            return Response(serialized_audit.data)
```
Comment (JeremyGibson) [inline api/views/message.py]: shouldn't there be and explicit status sent here?

**Review by JeremyGibson — CHANGES_REQUESTED** [review]

File: api/tests/test_signals.py (hunk 2 above)
Comment (michael-caktus) [reply api/tests/test_signals.py:7]: isn't it being used on line 10?

Comment (JeremyGibson) [reply api/tests/test_signals.py:7]: I think Colin's change makes it apply everywhere. 

> Autouse fixtures (xUnit setup on steroids)
> The class-level transact fixture is marked with autouse=true which implies that all test >methods in the class will use this fixture without a need to state it in the test function >signature or with a class-level usefixtures decorator.

Comment (copelco) [reply api/tests/test_signals.py:7]: ``mock_core_registry`` is used everywhere, but it needs to be imported here to check the values passed to the mock object

File: api/documents/message.py [hunk 4]
```diff
@@ -57,4 +58,4 @@ class MessageDocument(Document):
     class Django(object):
         model = Message
         # True disables auto-indexing
```
Comment (copelco) [inline api/documents/message.py]: I think we should just remove these commented out lines now

File: api/tests/test_signals.py (hunk 2 above)
Comment (michael-caktus) [reply api/tests/test_signals.py:7]: Maybe it's still useful to keep here so we can assert that it is called?

**Pull request merged**